src/design-patterns/MazeGame/door.hpp

```diff
@@ -0,0 +1,27 @@
+#ifndef DOOR_H
+#define DOOR_H
+
+#include "map_site.hpp"
+#include "room.hpp"
+
+class Door : public MapSite {
+ public:
+  Door() = delete;
+  Door(Room* r1, Room* r2);
+  ~Door() = default;
+
+  Door(const Door &) = delete;             // copy constructor
+  Door(Door &&) = delete;                  // move constructor
+  Door &operator=(const Door &) = delete;  // copy assignment
+  Door &operator=(Door &&) = delete;       // move assignment
+
+  void Enter() override;
+  Room *OtherSideFrom(Room *);
```

Comment:
사실 무슨 함수인지 모르겠음
Room의 내용이 변경되지 않아야 된다면 const 붙이자

```suggestion
  Room *OtherSideFrom(const Room *);
```